src/struct.c

```diff
@@ -631,6 +631,52 @@ Dsymbol *AggregateDeclaration::searchCtor()
     return s;
 }
 
+/***************************************
+ * Return true if aggregate is an InputRange.
+ */
+bool AggregateDeclaration::isInputRange()
+{
+    return (this->search(Loc(), Id::Ffront) &&
```

Comment:
Do I have to check

- that the return type `front()` is assignable to the element of the D darray and
- that empty returns `bool` and
- that `popFront()` takes no arguments and returns `void`?